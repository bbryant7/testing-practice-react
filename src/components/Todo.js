import React, {Component} from 'react';

class Todo extends Component {
    render() {
        const {todo, handleToggle} = this.props;

        return (
            <div className="Todo">
                <input type="checkbox" checked={todo.done} onChange={event => handleToggle(todo.id)}/>
                <span className={todo.done ? "todo-done" : "todo-not-done"} onClick={event => handleToggle(todo.id)}>{todo.text}</span>
            </div>
        )
    }
}

export default Todo;
