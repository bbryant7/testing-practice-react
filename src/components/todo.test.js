// src/components/Todo.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import Todo from "../components/Todo.js";

test('Todo renders correctly', () => {
    const tree = renderer.create(
       <Todo todo={{text: "Test", id: 1, done: false}} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
