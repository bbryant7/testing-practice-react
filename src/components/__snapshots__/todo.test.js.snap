// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo renders correctly 1`] = `
<div
  className="Todo"
>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="todo-not-done"
    onClick={[Function]}
  >
    Test
  </span>
</div>
`;
